cli/src/lib.rs

```diff
@@ -298,7 +307,8 @@ impl Iroha {
         }
         .start();
 
-        let snapshot_maker = SnapshotMaker::from_config(&config.snapshot, sumeragi.clone()).start();
+        let snapshot_maker =
+            SnapshotMaker::from_config(&config.snapshot, &sumeragi).map(SnapshotMaker::start);
```

Comment:
I like this change as it makes it clear from the code that `start` might not run without jumping to the `start` implementation